File: Examples/CSharp/Programming-Documents/Images/RemoveWatermark.cs
using System;
using System.Collections;
using System.IO;
using Aspose.Words;
using Aspose.Words.Tables;
using Aspose.Words.Fields;
using Aspose.Words.Layout;
using Aspose.Words.Drawing;
using System.Drawing;

namespace Aspose.Words.Examples.CSharp.Programming_Documents.Working_with_Images
{
    class RemoveWatermark
    {
        // ExStart:RemoveWatermark
        public static void Run()
        {          
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_WorkingWithImages();
            string fileName = "RemoveWatermark.docx";
            Document doc = new Document(dataDir + fileName);
            RemoveWatermarkText(doc);
            dataDir = dataDir + RunExamples.GetOutputFilePath(fileName);
            doc.Save(dataDir);
        }

        private static void RemoveWatermarkText(Document doc)
        {
            foreach (HeaderFooter hf in doc.GetChildNodes(NodeType.HeaderFooter, true))
            {
                foreach (Shape shape in hf.GetChildNodes(NodeType.Shape, true))
                {
                    if (shape.Name.Contains("WaterMark"))
                    {
                        shape.Remove();
                    }
                }
            }
        }
    }
    // ExEnd:RemoveWatermark
}

File: Examples/VisualBasic/VisualBasic/Programming-Documents/Images/RemoveWatermark.vb
Imports Aspose.Words.Drawing

Public Class RemoveWatermark
    ' ExStart:RemoveWatermark
    Public Shared Sub Run()
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_WorkingWithImages()
        Dim fileName As String = "RemoveWatermark.docx"
        Dim doc As New Document(dataDir & fileName)
        RemoveWatermarkText(doc)
        dataDir = dataDir & RunExamples.GetOutputFilePath(fileName)
        doc.Save(dataDir)
    End Sub

    Private Shared Sub RemoveWatermarkText(doc As Document)
        For Each hf As HeaderFooter In doc.GetChildNodes(NodeType.HeaderFooter, True)
            For Each shape As Shape In hf.GetChildNodes(NodeType.Shape, True)
                If shape.Name.Contains("WaterMark") Then
                    shape.Remove()
                End If
            Next
        Next
    End Sub
    ' ExEnd:RemoveWatermark
End Class
